model/SQLQuery.php

```diff
@@ -903,9 +903,13 @@ public function sql() {
 				else if(sizeof($join['filter']) == 1) $filter = $join['filter'][0];
 				else $filter = "(" . implode(") AND (", $join['filter']) . ")";
 
-				$table = strpos(strtoupper($join['table']), 'SELECT') ? $join['table'] : "\"" 
-					. $join['table'] . "\"";
-				$aliasClause = ($alias != $join['table']) ? " AS \"" . Convert::raw2sql($alias) . "\"" : "";
+				// Ensure tables are quoted, unless the table is actually a sub-select
+				$table = preg_match('/\bSELECT\b/i', $join['table'])
+					? $join['table']
+					: "\"{$join['table']}\"";
+				$aliasClause = ($alias != $join['table'])
+						? " AS \"{$alias}\""
```

Comment:
Changed because raw2sql is not the Ansi standard way to escape SQL identifiers.

http://savage.net.au/SQL/sql-99.bnf.html#delimited%20identifier